### PR TITLE
Get scanner preferences from OSP-OpenVAS scanner

### PR DESCRIPTION
**Pull request by timopollmeier**
The preferences of the OpenVAS scanner are expected to be in the
nvt_preferences table, so they have to be added there.

**Checklist**:

- Tests N/A
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: src/manage_sql_nvts.c [hunk 1]
```diff
@@ -1415,6 +1415,7 @@ manage_update_nvt_cache_osp (const gchar *update_socket)
 {
   osp_connection_t *connection;
   gchar *db_feed_version, *scanner_feed_version;
+  GSList *scanner_prefs;
```
Comment (mattmundell) [inline src/manage_sql_nvts.c]: This is only used in one block so it should be declared in that block.

File: src/manage_sql_nvts.c [hunk 2]
```diff
@@ -1496,6 +1497,65 @@ manage_update_nvt_cache_osp (const gchar *update_socket)
...
+        {
+          GString *prefs_sql;
+          GSList *point;
+          int first;
+
+          point = scanner_prefs;
+          first = 1;
+
+          osp_connection_close (connection);
+          prefs_sql = g_string_new ("INSERT INTO nvt_preferences (name, value)"
```
Comment (mattmundell) [inline src/manage_sql_nvts.c]: I think this needs to handle the case where the scanner preference exists already.

File: src/manage_sql_nvts.c [hunk 3]
```diff
@@ -1496,6 +1497,65 @@ manage_update_nvt_cache_osp (const gchar *update_socket)
                " VALUES ('checked_preferences', 1)"
                " ON CONFLICT (name) DO UPDATE SET value = EXCLUDED.value;");
         }
+
```
Comment (mattmundell) [inline src/manage_sql_nvts.c]: I think this should come before telling the main process to update its NVT cache.

File: src/manage_sql_nvts.c (hunk 1 above)
Comment (timopollmeier) [reply src/manage_sql_nvts.c]: I've moved the variable in daa969520825068f563c0a03752b730ba7e0c4ab.

File: src/manage_sql_nvts.c (hunk 3 above)
Comment (timopollmeier) [reply src/manage_sql_nvts.c]: Okay, I've changed it in 188e38bf9ea95129cc036d5c185823e9b25f8c8d.

File: src/manage_sql_nvts.c [hunk 4]
```diff
@@ -1496,6 +1497,65 @@ manage_update_nvt_cache_osp (const gchar *update_socket)
...
+          osp_connection_close (connection);
+          prefs_sql = g_string_new ("INSERT INTO nvt_preferences (name, value)"
+                                    " VALUES");
+          while (point)
+            {
+              osp_param_t *param;
+              gchar *quoted_name, *quoted_value;
+
+              param = point->data;
+              quoted_name = sql_quote (osp_param_name (param));
```
Comment (jjnicola) [inline src/manage_sql_nvts.c]: Here you should use the param_id instead of the param_name

File: src/manage_sql_nvts.c (hunk 2 above)
Comment (timopollmeier) [reply src/manage_sql_nvts.c]: This should be handled by the `ON CONFLICT (name) ...` clause and the I've run the enclosing if block multiple times without getting any duplicates.

File: src/manage_sql_nvts.c (hunk 4 above)
Comment (timopollmeier) [reply src/manage_sql_nvts.c]: I've changed it in bf4b700804dd4c35c02609d54cdd5dd18869adf4.

File: src/manage_sql_nvts.c (hunk 2 above)
Comment (mattmundell) [reply src/manage_sql_nvts.c]: Oh I missed that, nice.

**Pull request merged**